client/src/main/java/io/split/service/SplitHttpClientKerberosImpl.java

```diff
@@ -0,0 +1,220 @@
+
+import org.apache.hc.client5.http.classic.methods.HttpGet;
+import org.apache.hc.core5.http.Header;
+import org.apache.hc.core5.http.HttpEntity;
+import org.apache.hc.core5.http.HttpRequest;
+import org.apache.hc.core5.http.io.entity.EntityUtils;
+import org.apache.hc.core5.http.message.BasicHeader;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.*;
+import java.net.*;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+public class SplitHttpClientKerberosImpl implements SplitHttpClient {
+
+    private static final Logger _log = LoggerFactory.getLogger(SplitHttpClient.class);
+    private static final String HEADER_CACHE_CONTROL_NAME = "Cache-Control";
+    private static final String HEADER_CACHE_CONTROL_VALUE = "no-cache";
+    private static final String HEADER_API_KEY = "Authorization";
+    private static final String HEADER_CLIENT_KEY = "SplitSDKClientKey";
+    private static final String HEADER_CLIENT_MACHINE_NAME = "SplitSDKMachineName";
+    private static final String HEADER_CLIENT_MACHINE_IP = "SplitSDKMachineIP";
+    private static final String HEADER_CLIENT_VERSION = "SplitSDKVersion";
+
+    private final RequestDecorator _requestDecorator;
+    private final String _apikey;
+    private final SDKMetadata _metadata;
+
+    public static SplitHttpClientKerberosImpl create(RequestDecorator requestDecorator,
+                                                     String apikey,
+                                                     SDKMetadata metadata) throws URISyntaxException {
+        return new SplitHttpClientKerberosImpl(requestDecorator, apikey, metadata);
+    }
+
+    SplitHttpClientKerberosImpl(RequestDecorator requestDecorator,
+                                String apikey,
+                                SDKMetadata metadata) {
+        _requestDecorator = requestDecorator;
+        _apikey = apikey;
+        _metadata = metadata;
+    }
+
+    public synchronized SplitHttpResponse get(URI uri, FetchOptions options, Map<String, List<String>> additionalHeaders) {
+        HttpURLConnection getHttpURLConnection = null;
+        try {
+            getHttpURLConnection = (HttpURLConnection) uri.toURL().openConnection();
+            return _get(getHttpURLConnection, options, additionalHeaders);
+        } catch  (Exception e) {
+            throw new IllegalStateException(String.format("Problem in http get operation: %s", e), e);
+        } finally {
+            try {
+                if (getHttpURLConnection != null) {
+                    getHttpURLConnection.disconnect();
+                }
+            } catch (Exception e) {
+                _log.error(String.format("Could not close HTTP URL Connection: %s", e), e);
+            }
+        }
+    }
+    public SplitHttpResponse _get(HttpURLConnection getHttpURLConnection, FetchOptions options, Map<String, List<String>> additionalHeaders) {
```

Comment:
```suggestion
    public SplitHttpResponse get(HttpURLConnection getHttpURLConnection, FetchOptions options, Map<String, List<String>> additionalHeaders) {
```

Reply:
This is an internal call that takes HttpURLConnection to make the testing easier.
There is another get() call which takes URI as the first parameter

Reply:
So, can we find a better name instead of an underscore?

Reply:
Sure, will do